indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -185,3 +204,108 @@ class RHRegistrationEdit(RHManageRegistrationBase):
     def _process(self):
         return WPManageRegistration.render_template('management/registration_modify.html', self.event,
                                                     registration=self.registration)
+
+
+class RHRegistrationsActionBase(RHManageRegFormBase):
+    """Base class for classes performing actions on registrations"""
+
+    def _checkParams(self, params):
+        RHManageRegFormBase._checkParams(self, params)
+        ids = set(request.form.getlist('registration_id'))
+        self.registrations = (Registration
+                              .find(Registration.id.in_(ids), ~Registration.is_deleted)
+                              .with_parent(self.regform)
+                              .all())
+
+
+class RHRegistrationEmailRegistrants(RHRegistrationsActionBase):
+    """Send email to selected registrants"""
+
+    def _checkParams(self, params):
+        self._doNotSanitizeFields.append('from_address')
+        RHRegistrationsActionBase._checkParams(self, params)
+
+    def _send_emails(self, form):
+        for registration in self.registrations:
+            email_body = replace_placeholders('registration-email', form.body.data, registration=registration)
+            template = get_template_module('events/registration/emails/custom_email.html',
+                                           email_subject=form.subject.data, email_body=email_body)
+            email = make_email(to_list=registration.email, cc_list=form.cc_addresses.data,
+                               from_address=form.from_address.data, template=template, html=True)
+            send_email(email, self.event, 'Registration')
+        return len(self.registrations)
+
+    def _process(self):
+        form = EmailRegistrantsForm()
```

Comment:
What about having a default body? Check what I did for invitation emails.

This could be nice for example:

```html
{% extends 'emails/base.html' %}

{% block header_recipient -%}
    {first_name}
{%- endblock %}

{% block body %}
    {% block body_text %}
        We're contacting you because you registered for the event <strong>{event_title}</strong>.<br>
        <br><br>

        You can view your registration details on this page:<br>
        {link}
    {% endblock %}

    {% if session %}
        <br><br>
        Best regards<br>
        {{ session.user.full_name }}
    {% endif %}
{%- endblock %}

{% block footer_title %}Registration{% endblock %}
{% block footer %}
    {# Show default footer if sent from command line #}
    {% if not session.user %}
        {{ super() }}
    {% endif %}
{% endblock %}
```

That way people are encouraged to include the link to the registration details.